examples/crystal_volume/optimize_crystal_volyme.py

```diff
@@ -0,0 +1,213 @@
+from rlmolecule.sql.run_config import RunConfig
+from rlmolecule.tree_search.reward import RankedRewardFactory
+from rlmolecule.sql import Base, Session, digest, load_numpy_dict, serialize_ordered_numpy_dict
+from rlmolecule.sql.tables import GameStore, RewardStore, StateStore
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+parser = argparse.ArgumentParser(description='Run the crystal conduction ion volume optimization. ' +
+                                             'Default is to run the script locally')
+parser.add_argument('--config', type=str, help='Configuration file')
+parser.add_argument('--train-policy', action="store_true", default=False, help='Train the policy model only (on GPUs)')
+parser.add_argument('--rollout', action="store_true", default=False, help='Run the game simulations only (on CPUs)')
+
+args = parser.parse_args()
+
+run_config = RunConfig(args.config)
+run_id = run_config.run_id
+
+engine = run_config.start_engine()
+Base.metadata.create_all(engine, checkfirst=True)
+Session.configure(bind=engine)
+session = Session()
+
+
+class CrystalVolOptimizationProblem(CrystalProblem):
+    #def get_initial_state(self) -> CrystalState:
+    #    return CrystalState(rdkit.Chem.MolFromSmiles('C'), self._config)
+
+    def get_reward(self, state: CrystalState) -> (float, {}):
+        #if state.forced_terminal:
+        #    return qed(state.molecule), {'forced_terminal': True, 'smiles': state.smiles}
+        if state.terminal:
+            # TODO generate the pymatgen structure object for this decoration of the crystal structure
+            # and compute the volume of the conducting ions.
+            # For now, set a dummy reward which is the length of the string representation
+            return len(repr(state)), {'terminal': True, 'state_repr': repr(state)}
+        return 0.0, {'terminal': False, 'state_repr': repr(state)}
+
+
+def create_problem():
+    prob_config = run_config.problem_config
+
+    G = nx.DiGraph()
+    G2 = nx.DiGraph()
+    action_graph_file = "inputs/elements_to_compositions.edgelist.gz"
+    print(f"reading {action_graph_file}")
+    G = nx.read_edgelist(action_graph_file, delimiter='\t', data=False, create_using=G)
+    print(f'{G.number_of_nodes()} nodes, {G.number_of_edges()} edges')
+
+    action_graph2_file = "inputs/comp_type_to_decorations.edgelist.gz"
+    print(f"reading {action_graph2_file}")
+    G2 = nx.read_edgelist(action_graph2_file, delimiter='\t', data=False, create_using=G2)
+    print(f'{G2.number_of_nodes()} nodes, {G2.number_of_edges()} edges')
+
+    # also load the mapping from composition to composition type
+    working_dir = 'inputs'
+    df_comp = pd.read_csv(f'{working_dir}/compositions.csv.gz')
+    print(df_comp.head())
+    compositions = df_comp['composition'].to_list()
+    comp_types = set(df_comp['comp_type'].to_list())
+    comp_to_comp_type = dict(zip(df_comp['composition'], df_comp['comp_type']))
```

Comment:
Does this need to get spelled out in a csv? I would have thought that getting the composition type (i.e., _1_1_4) from a composition (i.e., Li1Sc1F4) would be pretty straightforward if the composition strings are reasonable

Reply:
Yes that is true. If we assume the composition convention won't change, then it's not that hard.